R/indexing.R

```diff
@@ -54,6 +53,63 @@
 # - G[1:3,2,eid=TRUE]
 #               create an edge sequence
 
+get_adjacency_submatrix <- function(x, i = NULL, j = NULL, attr = NULL, sparse = TRUE) {
+  # If i or j is NULL, assume all nodes
+  if (is.null(i)) i <- seq_len(vcount(x))
+  if (is.null(j)) j <- seq_len(vcount(x))
+
+  # Handle duplicates
+  unique_i <- unique(i)
+  unique_j <- unique(j)
+
+  # Create mapping between to unique values
+  i_map <- match(i, unique_i)
+  j_map <- match(j, unique_j)
+
+  adj <- adjacent_vertices(x, unique_i, mode = "out")
+
+  edge_list <- cbind(rep(unique_i, sapply(adj, length)), unlist(adj))
```

Comment:
This is a mouthful, can we extract the `rep()` call to a variable?

```suggestion
  edge_list <- cbind(rep(unique_i, map_int(adj, length)), unlist(adj))
```